brigade-api/cmd/brigade-api/filter.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"log"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	restful "github.com/emicklei/go-restful"
+)
+
+var (
+	green        = string([]byte{27, 91, 57, 55, 59, 52, 50, 109})
+	white        = string([]byte{27, 91, 57, 48, 59, 52, 55, 109})
+	yellow       = string([]byte{27, 91, 57, 55, 59, 52, 51, 109})
+	red          = string([]byte{27, 91, 57, 55, 59, 52, 49, 109})
+	blue         = string([]byte{27, 91, 57, 55, 59, 52, 52, 109})
+	magenta      = string([]byte{27, 91, 57, 55, 59, 52, 53, 109})
+	cyan         = string([]byte{27, 91, 57, 55, 59, 52, 54, 109})
+	reset        = string([]byte{27, 91, 48, 109})
+	disableColor = false
+)
+
+var logger *log.Logger = log.New(os.Stdout, "", 0)
+
+// NCSACommonLogFormatLogger Create a filter that produces log lines
+// according to the Common Log Format, also known as the NCSA standard.
+// Coloring inspired by gin
+func NCSACommonLogFormatLogger() restful.FilterFunction {
+	return func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		var username = "-"
+		if req.Request.URL.User != nil {
+			if name := req.Request.URL.User.Username(); name != "" {
+				username = name
+			}
+		}
+		chain.ProcessFilter(req, resp)
+		ip, _, err := net.SplitHostPort(strings.TrimSpace(req.Request.RemoteAddr))
+		if err != nil {
+			return
+		}
+		var statusColor, methodColor, resetColor string
+		methodColor = colorForMethod(req.Request.Method)
```

Comment:
Can you add a check for ANSI support.  Here is the one from gin https://github.com/gin-gonic/gin/blob/master/logger.go#L61-L63